cachi2/interface/cli.py

```diff
@@ -432,30 +432,50 @@ def merge_sboms(
 
     first to produce SBOMs to merge.
     """
-    sboms_to_merge: List[Union[Sbom, SPDXSbom]] = []
+    sboms_to_merge: List[Union[SPDXSbom, Sbom]] = []
     for sbom_file in sbom_files_to_merge:
+        sbom_dict = json.loads(sbom_file.read_text())
+        # Remove extra fields which are not in Sbom or SPDXSbom models
+        # Both SBom and SPDXSBom models are only subset of cyclonedx and SPDX specifications
+        # Therefore we need to make sure only fields accepted by the models are present
+        for key in list(sbom_dict.keys()):
+            if key not in Sbom.model_fields and key not in SPDXSbom.model_fields:
+                sbom_dict.pop(key)
         try:
-            if sbom_type == SBOMFormat.cyclonedx:
-                sboms_to_merge.append(Sbom.model_validate_json(sbom_file.read_text()))
-            else:
-                sboms_to_merge.append(SPDXSbom.model_validate_json(sbom_file.read_text()))
+            sboms_to_merge.append(Sbom(**sbom_dict))
         except pydantic.ValidationError:
-            raise UnexpectedFormat(f"{sbom_file} does not appear to be a valid Cachi2 SBOM.")
+            try:
+                sboms_to_merge.append(SPDXSbom(**sbom_dict))
+            except pydantic.ValidationError:
+                raise UnexpectedFormat(f"{sbom_file} does not appear to be a valid Cachi2 SBOM.")
 
     if sbom_type == SBOMFormat.cyclonedx:
+        cyclonedx_sboms_to_merge = []
+        for _sbom in sboms_to_merge:
+            if not isinstance(_sbom, Sbom):
+                cyclonedx_sboms_to_merge.append(_sbom.to_cyclonedx())
```

Comment:
I was thinking about this and I'm not sure if that good idea regarding to multiarch sboms. I haven't seen one yet, but I assume it will contain hierarchy like: root -> virtual-arch-package -> [ arch packages, ... ]. That would be lost if converted to spdx, so I think we should avoid any conversion if it's not necessary

Reply:
FWIW, Cachi2 does not produce multiarch SBOMs, so we don't need to support merging those. We can update the merging script once that changes, but I don't see it happening in the near future.